### ISSUE TITLE
ltrTreeLayouter gravitates leaves to the right

### ISSUE THREAD

**Issue by lightpriest** [Netflix/vizceral#48]
tl;dr: I was going through the source code and commits and couldn't find if the leaves gravitating to the right is desired or not.

It's actually more of a question. I created a graph and I noticed that on the region view (ltrTreeLayout), the leaves gravitate to the right. A render is better here (labels redacted):
![screenshot from 2017-01-16 19-03-23-circled](https://cloud.githubusercontent.com/assets/351445/22183498/8b2e4722-e0c8-11e6-8803-5f14a51102ba.png)

The leaves in question are circled in red. I would expect them to appear on +1 level of their parent. I modified the `ltrTreeLayouter.js` code to produce the following for the same `traffic.json`:
![screenshot from 2017-01-16 18-51-25-circled](https://cloud.githubusercontent.com/assets/351445/22183585/4b7e38c4-e0ca-11e6-9ee6-dac7353f5772.png)

My first assumption, as for the intuitive aspect of vizceral, is that it's a matter of personal preference. So I guess my question is, is it on purpose? And if it is, is there a way to override the rendering functions with our own (instead of changing vizceral's source)? If it's not, I'll happily send a PR with the modification.

**Comment by aSqrd-eSqrd** [comment]
@lightpriest,  It would be helpful to have a copy of the traffic JSON file you are using.  Could you sanatize a copy and add it as an attachment.

I recently found a bug in related to the ltrTreeLayout that involves a bug in the dfs subfunction of dfsFas in the acyclicFAS.js.  It is a one line fix, to make it more like the dfs function in ranker.js.   I just haven't gotten to submitting an issue with a fix... tomorrow hopefully.  I also don't know if it fixes anything but it stops a bunch of errors due to attempting to access an "undefined" property, but you don't usually see them since it is encapsulated in the web-worker.

**Comment by jrsquared** [comment]
@lightpriest Yes, that was on purpose.  We wanted to see all all of the leaf nodes (a.k.a. services without dependencies) on the far right. You are able to [specify your own layout function](https://github.com/Netflix/Vizceral/wiki/How-to-Use) if you want.  You can also add a new layout class before you load your data.

You just have to add your layout class to the [layouts object](https://github.com/Netflix/vizceral/blob/master/src/vizceral.js#L153) on the vizceral object.

The layout class has to have a function with the signature `run (graph, dimensions, layoutComplete)`.  You can see the existing layout classes as examples.

Let me know if that works for you.

**Comment by lightpriest** [comment]
@aSqrd-eSqrd Sure. [traffic.json.zip](https://github.com/Netflix/vizceral/files/738985/traffic.json.zip)

**Comment by anrub** [comment]
@lightpriest how did you change the layouting? I'm searching for this change, maybe you could share the patch for this :+1:

**Comment by lightpriest** [comment]
@anrub Would you believe me if I say I can't find it? :(
